### PR TITLE
feat(plot_ripley_l): Return a dataframe of the line plots

### PR DESCRIPTION
**Pull request by georgezakinih**

### CODE REVIEW

**Review by fangliu117 — COMMENTED** [review]
In general, the function code is logically clear, and the tests cover a wide range of scenarios, including simulations, multiple regions etc,  and includes edge cases. Only some minor comments are listed besides the lines for improvement.

File: src/spac/visualization.py [hunk 1]
```diff
@@ -1933,4 +1952,8 @@ def plot_ripley_l(
     ax.set_xlabel("Radii (pixles)")
```
[the remark this answers is not in the document]
Comment (georgezakinih) [reply src/spac/visualization.py:1952]: Thanks Fang. Fixed.

File: src/spac/visualization.py [hunk 2]
```diff
@@ -1909,6 +1916,18 @@ def plot_ripley_l(
...
+        for _, stat_row in l_stat_data.iterrows():
+            plot_data.append({
+                'region': region,
+                'radius': stat_row['bins'],
+                'ripley(radius)': stat_row['stats'],
+                'region_area': area,
+                'n_center': n_center,
+                'n_neighbor': n_neighbors,
+            })
+
```
[the remark this answers is not in the document]
Comment (georgezakinih) [reply src/spac/visualization.py:1930]: Thanks Fang, fixed.

File: src/spac/visualization.py [hunk 3]
```diff
@@ -1909,6 +1916,18 @@ def plot_ripley_l(
             ax=ax,
             **kwargs)
 
+        # Prepare plotted data to return if return_df is True
```
[the remark this answers is not in the document]
Comment (georgezakinih) [reply src/spac/visualization.py:1919]: Thanks Fang,  I can't split the legend on multiple lines as the legend box will become very large.
